contracts/integrations/test/utils/function_assertions.ts

```diff
@@ -9,18 +12,27 @@ export interface ContractWrapperFunction extends ContractGetterFunction {
     awaitTransactionSuccessAsync?: (...args: any[]) => PromiseWithTransactionHash<TransactionReceiptWithDecodedLogs>;
 }
 
-export interface Condition {
-    before: (...args: any[]) => Promise<any>;
-    after: (beforeInfo: any, result: Result, ...args: any[]) => Promise<any>;
-}
-
 export interface Result {
     data?: any;
-    receipt?: TransactionReceiptWithDecodedLogs;
     success: boolean;
+    receipt?: TransactionReceiptWithDecodedLogs;
+}
+
+export interface Condition {
```

Comment:
Since these are so core to the testing framework it could also be good to add descriptions/usage for these interfaces.